src/htcp.cc

```diff
@@ -1588,7 +1589,8 @@ htcpClear(StoreEntry * e, HttpRequest * req, const HttpRequestMethod &, CachePee
     stuff.S.uri = uri.c_str();
     stuff.S.version = vbuf;
     if (reason != HTCP_CLR_INVALIDATION) {
-        HttpStateData::httpBuildRequestHeader(req, e, nullptr, &hdr, flags);
+        const auto peer = cbdataReferenceValid(p) ? p : nullptr;
+        HttpStateData::httpBuildRequestHeader(req, e, nullptr, &hdr, peer, flags);
```

Comment:
We should not check `p` validity (here or at all) because we dereference `p` above. I suggest using without checking in this case.

Reply:
Ok.